Source/WebCore/dom/Document.cpp

```diff
@@ -8067,6 +8068,17 @@ void Document::dispatchPagehideEvent(PageshowEventPersistence persisted)
     dispatchWindowEvent(PageTransitionEvent::create(eventNames().pagehideEvent, persisted == PageshowEventPersistence::Persisted), this);
 }
 
+bool Document::resolveViewTransitionRule()
```

Comment:
```suggestion
bool Document::shouldPerformCrossDocumentViewTransition()
```

I'm not sure what you're planning to use this for, but the name could be more descriptive.

(Maybe that should take an `origin` parameter and check if it's the same origin too)

Reply:
Matt pointed out this matches a spec algorithm, this is fine as it is, with a spec link added.